mediasms.go

```diff
@@ -0,0 +1,499 @@
+	Password string
+	Prefix   string
+}
+
+// SendRequest struct
+type SendRequest struct {
+	Smsid         string `json:"smsid"`
+	Smstitle      string `json:"smstitle"`
+	Smstext       string `json:"smstext"`
+	Mobilenumber  string `json:"mobilenumber"`
+	Originalurl   string `json:"originalurl"`
+	Originalurl2  string `json:"originalurl2"`
+	Originalurl3  string `json:"originalurl3"`
+	Originalurl4  string `json:"originalurl4"`
+	Status        string `json:"status"`
+	Returnsms     string `json:"returnsms"`
+	Waitreturnsms string `json:"waitreturnsms"`
+	Type          string `json:"type"`
+}
+
+// APIResponse struct
+type APIResponse struct {
+	StatusCode  string
+	Name        string
+	Description string
+}
+
+// ResultCode from media4u
+type ResultCode struct {
+	Result string `json:"result"`
+}
+
+// WebHook from media4u
+type WebHook struct {
+	Mobilenumber      string
+	Status            string
+	Smsid             string
+	Returnsms         string
+	Raitreturnsms     string
+	Returnsmsdatetime string
+	Replyid           string
+	Senderid          string
+}
+
+const smsurl = "https://www.sms-console.jp/"
+
+// Send request to media4u
+func (c Client) Send(messageID string, val SendRequest) (APIResponse, error) {
+	smsID := createSMSID(c.Prefix, messageID)
+
+	val.Smsid = smsID
+
+	jsonValue, _ := json.Marshal(val)
+	client := &http.Client{}
+	req, err := http.NewRequest("POST", smsurl+"api/", bytes.NewBuffer(jsonValue))
+	req.SetBasicAuth(c.Username, c.Password)
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := client.Do(req)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	defer resp.Body.Close()
+
+	body, _ := ioutil.ReadAll(resp.Body)
+	var sendResult ResultCode
+	json.Unmarshal(body, &sendResult)
+
+	var res = sendResultsMapper[sendResult.Result]
+
+	results := APIResponse{
+		StatusCode:  sendResult.Result,
+		Name:        res["name"],
+		Description: res["description"],
+	}
+
+	return results, err
+}
+
+// GetStatus of a sent sms
+func (c Client) GetStatus(messageID string) (APIResponse, error) {
+	smsID := createSMSID(c.Prefix, messageID)
+
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", smsurl+"api5/?smsid="+smsID, nil)
+	req.SetBasicAuth(c.Username, c.Password)
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := client.Do(req)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	s := string(body)
+	t := strings.Split(s, "\n")
+
+	if t[0] == "200" {
+		results := APIResponse{
+			StatusCode:  "200",
+			Name:        "Success",
+			Description: translationMap[t[1]],
+		}
+		return results, err
+	}
+
+	var getResult ResultCode
+	json.Unmarshal(body, &getResult)
+
+	res := getResultsMapper[getResult.Result]
+
+	results := APIResponse{
+		StatusCode:  getResult.Result,
+		Name:        res["name"],
+		Description: res["description"],
+	}
+
+	return results, err
+}
+
+func createSMSID(prefix, messageID string) string {
```

Comment:
`prefix` seems to miss the type

Reply:
If all parameters are the same, you only need to give the type once. So in this example, both `prefix` and `messageID` are strings. So we don't need to declare it each time.